frontend/ui/widget/inputdialog.lua

```diff
@@ -1,3 +1,40 @@
+--[[--
+Widget for taking user input.
+
+Example:
+
+    local _ = require("gettext")
+    local UIManager = require("ui/uimanager")
+    local sample_input
+    local saveHandler = function()
+        print('Got user input:', sample_input:getInputText())
+    end
+    sample_input = InputDialog:new{
+        title = _("Dialog title"),
+        input = "default value",
+        input_hint = "hint text",
+        input_type = "text",
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    callback = function()
+                        UIManager:close(sample_input)
+                    end,
+                },
+                {
+                    text = _("Save"),
+                    callback = saveHandler,
+                },
+            }
+        },
+        enter_callback = saveHandler,
```

Comment:
I remember in WPF or visual basic, there is a default button in one form, which responds enter keypress. We can use similar approach, i.e. a default_button instead of enter_callback, with default value as 1. Then we won't need to set enter_callback for most of the following logic. The advantage is, enter behavior should always be consistent with one of the buttons, otherwise devices with touch screen only would loss a certain functionality.